core/capabilities/webapi/trigger/trigger.go

```diff
@@ -0,0 +1,299 @@
+		connector:           connector,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// Iterate over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload webapicapabilities.TriggerRequestPayload) {
+	// Pass on the payload with the expectation that that is acceptable format for the executor
+	wrappedPayload, _ := values.WrapMap(payload)
+	hasMatchedAWorkflow := false
+	var response webapicapabilities.TriggerResponsePayload
+
+	for _, trigger := range h.registeredWorkflows {
+		topics := payload.Topics
+		// empty topics means all topics
+		if len(topics) == 0 {
+			topics = []string{}
+			for k := range trigger.allowedTopicsMap {
+				topics = append(topics, k)
+			}
+		}
+
+		for _, topic := range topics {
+
+			if trigger.allowedTopicsMap[topic] {
+				hasMatchedAWorkflow = true
+
+				if !trigger.allowedSendersMap[sender.String()] {
+					h.lggr.Errorw("Unauthorized Sender")
+					h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "Unauthorized Sender"})
+					return
+				}
+				if !trigger.rateLimiter.Allow(body.Sender) {
+					h.lggr.Errorw("request rate-limited")
+					h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "request rate-limited"})
+					return
+				}
+
+				TriggerEventID := body.Sender + payload.TriggerEventID
+				tr := capabilities.TriggerResponse{
+					Event: capabilities.TriggerEvent{
+						TriggerType: webapicapabilities.TriggerType,
+						ID:          TriggerEventID,
+						Outputs:     wrappedPayload,
+					},
+				}
+
+				trigger.ch <- tr
+				response = webapicapabilities.TriggerResponsePayload{Status: "ACCEPTED"}
+				// Sending n topics that match a workflow with n allowedTopics, can only be triggered once.
+				break
+			}
+		}
+	}
+	if !hasMatchedAWorkflow {
+		h.lggr.Errorw("No Matching Workflow Topics")
+		response = webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "No Matching Workflow Topics"}
+	}
+	_ = h.sendResponse(ctx, gatewayID, body, response)
+
+}
+
+// https://gateway-us-1.chain.link/web-trigger
+//   {
+//     jsonrpc: "2.0",
+//     id: "...",
+//     method: "web-trigger",
+//     params: {
+//       signature: "...",
+//       body: {
+//         don_id: "workflow_123",
+//         payload: {
+//           trigger_id: "web-trigger@1.0.0",
+//           trigger_event_id: "action_1234567890",
+//           timestamp: 1234567890,
+//           topics: ["daily_price_update"],
+//           params: {
+//             bid: "101",
+//             ask: "102"
+//           }
+//         }
+//       }
+//     }
+//   }
+
+// from Web API Trigger Doc
```

Comment:
These descriptions should live on the type https://github.com/smartcontractkit/chainlink/pull/14580/files#diff-c7c1690d0593c367c96ae8a5527628c12ba67fa7685688ab64c89d5feca95528R25 instead of here